cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/AbstractHttpHandlerDelegator.java

```diff
@@ -118,11 +140,60 @@ protected final DelayedHttpServiceResponder wrapResponder(HttpResponder responde
   protected final BodyConsumer wrapContentConsumer(HttpContentConsumer consumer,
                                                    DelayedHttpServiceResponder responder,
                                                    TransactionContext txContext) {
+    Preconditions.checkState(!responder.hasBufferedResponse(),
+                             "Response cannot be sent when using HttpContentConsumer.");
```

Comment:
It tries to capture this misuse.

```java
HttpContentConsumer handle(HttpServiceRequest request, HttpServiceResponder responder) {
  responder.sendStatus(200);
  return new HttpContentConsumer() {
     ...
  };
}
```

because response should be sent from the `onFinish` method in the `HttpContentConsumer`. Any recommendation on what the error message should be?